fix: preserve fetch_links across chained find() and find_one()

scarlet2131 opened the pull request:
### What

This PR fixes a bug where `fetch_links=True` was not preserved when chaining multiple `.find()` or `.find_one()` calls.

### Why

As reported in [issue #1182](https://github.com/BeanieODM/beanie/issues/1182), chaining `.find()` caused the `fetch_links` flag to be dropped, resulting in unresolved linked documents unless manually re-added.

### Fix

This change ensures that `fetch_links`, `nesting_depth`, and related flags are retained across chained queries by setting them explicitly during `.find()` reuse.

### Test

Reproduced the issue locally using a minimal example with linked models and validated that the bug no longer occurs after this fix.

Before the fix : 

<img width="1438" alt="Screenshot 2025-06-09 at 2 05 41 PM" src="https://github.com/user-attachments/assets/f4971248-5853-44d0-8940-3f02719a0dd9" />

After the fix : 

<img width="1440" alt="Screenshot 2025-06-09 at 2 05 00 PM" src="https://github.com/user-attachments/assets/540d37c3-ce0f-4459-8593-10aad7ba2461" />

staticxterm commented:
Hi @scarlet2131,
thank you very much for the PR.
Could you please mention the usage and this behavior in the Beanie documentation pages as well, for the find() examples:
https://beanie-odm.dev/tutorial/finding-documents/
https://github.com/BeanieODM/beanie/blob/main/docs/tutorial/find.md

What is worrying me is this "limitation" description as mentioned here:
https://beanie-odm.dev/tutorial/relations/#limitations
Could you please write some tests for the chaining examples to have it covered with tests. Also, to see what error is raised when chaining e.g. `find(..., fetch_links=True).delete()` as mentioned previously.

scarlet2131 commented:
Hi @adeelsohailahmed, @staticxterm, and @CAPITAINMARVEL,

I have added tests to confirm that using fetch_links=True with chained .find() now works correctly when followed by .delete() and .update() operations.

These tests are added in:

test_find.py for basic chaining and deletion

test_relations.py to verify behavior with linked documents and to show that deletion and update now work as expected

The tests cover:

Chaining multiple .find() calls with fetch_links=True

Deleting and updating documents in those chained queries

Ensuring that the linked documents are correctly handled

I also updated the documentation in docs/tutorial/find.md to show an example of how this chaining works with fetch_links.

All tests are passing locally. Please let me know if there's anything else I should do.
